src/run/parser/source_live_timer.rs

```diff
@@ -0,0 +1,106 @@
+//! Provides the parser for the SourceLiveTimer splits files.
+
+#![allow(non_snake_case)]
+
+use std::io::Read;
+use std::result::Result as StdResult;
+use serde_json::de::from_reader;
+use serde_json::Error as JsonError;
+use {GameTime, Run, Segment, Time, TimeSpan};
+
+quick_error! {
+    /// The Error type for splits files that couldn't be parsed by the
+    /// SourceLiveTimer Parser.
+    #[derive(Debug)]
+    pub enum Error {
+        /// Failed to parse JSON.
+        Json(err: JsonError) {
+            from()
+        }
+    }
+}
+
+/// The Result type for the SourceLiveTimer parser.
+pub type Result<T> = StdResult<T, Error>;
+
+#[derive(Deserialize)]
+struct Splits {
+    Category: String,
+    RunName: Option<String>,
+    Splits: Option<Vec<Split>>,
+}
+
+#[derive(Deserialize)]
+struct Split {
+    Map: String,
+    Name: Option<String>,
+    #[allow(dead_code)]
+    Ticks: Option<u64>,
+    Segment: Option<u64>,
+    BestSegment: Option<u64>,
+}
+
+fn time_span_from_ticks(category_name: &str, ticks: u64) -> TimeSpan {
+    let seconds = if category_name.starts_with("Portal 2") {
+        ticks as f64 / 30.0
+    } else {
+        // Game is either Portal or Half Life 2
+        ticks as f64 / 66.6666666666667
+    };
+
+    TimeSpan::from_seconds(seconds)
+}
+
+/// Attempts to parse a SourceLiveTimer splits file.
+pub fn parse<R: Read>(source: R) -> Result<Run> {
+    let mut run = Run::new();
+    let splits: Splits = from_reader(source)?;
+
+    if splits.Category.starts_with("Portal 2") {
+        run.set_game_name("Portal 2");
+    } else if splits.Category.starts_with("Portal") {
+        run.set_game_name("Portal");
+    } else if splits.Category.starts_with("Half Life 2") {
+        run.set_game_name("Half Life 2");
+    }
+
+    if let Some(run_name) = splits.RunName {
+        if run_name != splits.Category {
+            run.set_category_name(run_name);
```

Comment:
This seems to suggest that the run name is more likely to be the category name than the .Category field? Can you elaborate on that?

Reply:
From looking at the source for SourceLiveTimer it appears that RunName can be changed, where as The CategoryName references a static name that's set by when making the splits, the first phrase of the String is the name of the game. If the RunName isn't changed by the user it defaults to the same as the CategoryName.

Reply:
alright :)